src/core/xref-db.js

```diff
@@ -52,10 +51,22 @@ async function resolveFromCache(keys, cache) {
 }
 
 /**
+ * Get last updated timestamp from server and bust cache based on that. This way, we
+ * prevent dirty/erroneous/stale data being kept on a client (which is possible
+ * if we use a `MAX_AGE` based caching strategy).
  * @param {number} cachedTime
  */
-function isBustedCache(cachedTime) {
-  return Date.now() - cachedTime > CACHE_MAX_AGE;
+async function isBustedCache(cachedTime) {
+  const url = new URL("meta/version", API_URL);
+  try {
+    const res = await fetch(url);
+    if (!res.ok) throw new Error(res.statusText);
+    const lastUpdated = parseInt(await res.text(), 10);
+    return lastUpdated > cachedTime;
+  } catch {
```

Comment:
Probably best to just console.error(err) here ... silent exceptions "no es bueno"

Reply:
On a second thought, I think this try-catch is redundant. we can return `false` right away on `res.ok` instead of throwing and later catching

Reply:
sounds good.